### PR TITLE
Pneumatic Cannon Nerf

### LINKED ISSUE #7046: pneumatic cannons are OP as fuck now. The new syringe gun
Firing a pneumatic cannon loaded with food at a person will cause them to take a bite if it hits. 

Here's why it's better than syringe gun

+don't need syringes to hold the malicious chems, any piece of food can do including vending machine foods
+less suspicious
+can go through hardsuits
+clown can use them (maybe, not tested)
+better ammo storage

-less chemicals transfered, but for some chems it doesn't take much in the first place

BALANCE DISCUSSION WHEN. SURE IT'S FUN BUT IT ONLY TAKES ONE SHITTER (me) TO ABUSE THIS

### LINKED ISSUE #11621: Pneumatic Cannon is pretty overpowered
Seeing it one shot ANY and EVERYONE is pretty crazy, especially for Wizards/Lings/Vamps.

It doesn't seem to be hard to set up too.

### LINKED ISSUE #12400: Time to fix the pneumatic cannon
Fact that you can fire glass shard and do 300+ damage is dumb.

Fix how the damage works or put it in a lock box

### PR DESCRIPTION
**Pull request by JMWTurner**
Fixes #7046
Fixes #11621
Fixes #12400

:cl:
- tweak: Wearing a mask or an helmet will protect you from the pneumatic cannon's force feeding effect.
- tweak: Clumsy players now have a 50 percent chance to fire at themselve with the pneumatic cannon.
- tweak: halved the pneumatic cannon's projectile speed cap. A glass shard fired using half an oxygen tank's full pressure capacity can still instacrit a naked target, but not outright kill it.

### CODE REVIEW

**Comment by JMWTurner** [comment]
Fixed compile errror. Phew, I thought the pneumatic cannon was a gun all this time.

**Comment by ghost** [comment]
Can you test some items in it and report their damage? 
The main contenders for it were: hatchet, shard, skull, screwdriver and null rod. 
Just wanna see what the damage is like now.

**Comment by JMWTurner** [comment]
@whorena the speed was previously capped to 80, it is now capped to 40.

The formula for damage on mobs is "(throwforce x (speed / 5)"

So a shard with a throwforce of 15 could previously deal up to 240 damage in one hit. Now only 120. I'll test with the other ones.

**Comment by JMWTurner** [comment]
@whorena

Alright, here are the test's results, each hit was done on a different target:
- Filled oxy tank to the brim
- Loaded a **shard**, set pressure release to **50**%
- First hit = 135 brute damage, target in crit
- Second hit = 135 brute damage, target in crit
- Third hit = 81 brute damage
- Fourth hit = 48 brute damage
- Fifth hit = 16 brute damage

second tests
- Filled oxy tank to the brim
- Loaded a **shard**, set pressure release to **25**%
- First hit = 135 brute damage, target in crit
- Second hit = 114 brute damage, target in crit
- Third hit = 89 brute damage
- Fourth hit = 71 brute damage
- Fifth hit = 41 brute damage

**Comment by JMWTurner** [comment]
tomahawk, 50%
- 1st: 135 damage
- 2nd: 81 damage
- 3rd: 48 damage
- 4th: 16 damage
- 5th: 8 damage

tomahawk, 25%
- 1st: 66 damage
- 2nd: 64 damage
- 3rd: 37 damage
- 4th: 43 damage (huh?)
- 5th: 20 damage

**Comment by JMWTurner** [comment]
skull, 50%
- 1st: 95 damage
- 2nd: 95 damage
- 3rd: 59 damage
- 4th: 21 damage
- 5th: 10 damage

skull, 25%
- 1st: 95 damage
- 2nd: 81 damage
- 3rd: 64 damage
- 4th: 37 damage
- 5th: 13 damage

**Comment by JMWTurner** [comment]
I think I'll stop there, looks like glass shard really are the most dangerous projectile after all.

**Comment by Shadowmech88** [comment]
> halved the pneumatic cannon's projectile speed cap.

Why nerf the pneumatic cannon itself if the only thing broken about it is firing glass shards out of it?
Glass shards deal so much damage because they've got a throwforce of 15 and a w_class of 1.

Why not just nerf the throwforce of glass shards instead?

**Comment by ghost** [comment]
Alright it's not as bad as I thought but I agree with Shadow a little.
The w_class_tiny items are the major problem, theyre abundant and strong in the thing. I previously suggested we limited them out like the blunderbuss has. This, plus the half speedcap nerf, means that tomohawks and hatchets become the deadliest thing you can stuff in it (I think they're w_class_small) without completely devaluing it as a powerful weapon.

**Comment by JMWTurner** [comment]
Remember that whenever you see 135 damage it means that it's been capped and the original damage was higher.

File: code/modules/projectiles/guns/projectile/pneumatic.dm [hunk 1]
```diff
@@ -127,17 +133,17 @@
 		return 0
 
 	var/obj/item/object = contents[1]
-	var/speed = ((fire_pressure*tank.volume)/object.w_class)/force_divisor //projectile speed.
-	if(speed>80)
-		speed = 80 //damage cap.
+	var/speed = min(40,((fire_pressure*tank.volume)/object.w_class)/force_divisor)
```
Comment (ComicIronic) [inline code/modules/projectiles/guns/projectile/pneumatic.dm]: Needs a define.

File: code/modules/projectiles/guns/projectile/pneumatic.dm [hunk 2]
```diff
@@ -127,17 +133,17 @@
...
 
 	user.visible_message("<span class='danger'>[user] fires [src] and launches [object] at [target]!</span>","<span class='danger'>You fire [src] and launch [object] at [target]!</span>")
 
 	src.remove_from_storage(object,user.loc)
 	object.throw_at(target,10,speed)
 
 	if(istype(object,/obj/item/weapon/reagent_containers/food/snacks) && ishuman(target) && object.Adjacent(target))
-		object.attack(target,target) //This way it is instant
+		var/mob/living/carbon/human/victim = target
+		if(!istype(victim.get_item_by_slot(slot_head), /obj/item/clothing/head/helmet) && !istype(victim.get_item_by_slot(slot_wear_mask), /obj/item/clothing/mask))
```
Comment (ComicIronic) [inline code/modules/projectiles/guns/projectile/pneumatic.dm]: You should check for mouth coverage instead of this.

**Comment by FudgePucker** [comment]
Would a shard of glass even do that much damage realistically?
A bullet to the face made out of lead moving past the sound barrier can't crit in one hit. But a shard of glass just fucking destroys someone? Wouldn't it shatter into dust before it reached a speed that could do more damage than a bullet?

**Comment by clusterfack** [comment]
You've been watching too many movies, shards of glass fucking kill people when used as shrapnel or when glass ceilings get shattered.

**Comment by Shadowmech88** [comment]
Yeah, but the question is "do they kill better than bullets do?"

Because being shot in the head by a bullet doesn't even kill in one hit in SS13, so why should a shard of glass to the head?

**Comment by clusterfack** [comment]
I admit that is fairly excessive

**Comment by JustSumBody** [comment]
How am I supposed to cuck manifested armor ghosts now

File: code/modules/projectiles/guns/projectile/pneumatic.dm (hunk 1 above)
Comment (ComicIronic) [reply code/modules/projectiles/guns/projectile/pneumatic.dm]: (for the speed cap)

**Comment by ComicIronic** [comment]
Is this going to be fixed?

**Comment by JMWTurner** [comment]
I'll take care of it this week end.

**Comment by JMWTurner** [comment]
there.

**Comment by ComicIronic** [comment]
Fine from the code end, I'm leaving this up to a playing collab to merge.

**Comment by Killette2** [comment]
Remember shards can still be "sharpened" and their force changed (albeit not controllably). This sharpness carries over to what they turn into, including spears. Unsure how this affects their throwforce? Don't remember that part of the code.

**Comment by JMWTurner** [comment]
@Killette2 Throwing a shard randomly sharpens it indeed, however this only affects its force, not its throwforce (which makes no sense whatsoever, but whatever).

**Pull request merged**